File: 2015/1/solution.java
import java.util.Scanner;

public class NotQuiteLisp {
    static Scanner sc = new Scanner(System.in);

    public static void main(String[] args) throws Exception {
        char[] input = sc.nextLine().trim().toCharArray();
        int floor = 0, basement = 0;

        for (int i = 0; i < input.length; i++) {
            floor += input[i] == '(' ? 1 : -1;
            if (floor == -1 && basement == 0) {
                basement = i + 1;
            }
        }

        System.out.println(floor);
        System.out.println(basement);
    }
}

File: 2015/1/solution.cpp
#include <iostream>
#include <string>

using namespace std;

int main() {
    int floor = 0;
    int basement = 0;

    string input;
    cin >> input;

    for (int i = 0; i < input.length(); i++) {
        floor += input[i] == '(' ? 1 : -1;
        if (floor == -1 && basement == 0) {
            basement = i + 1;
        }
    }

    cout << floor << endl;
    cout << basement << endl;

    return 0;
}